File: code/community/1304605478935859208/deepgram-speech-to-text-aws-lambda-websockets.py
import os
from deepgram import Deepgram
import asyncio

DEEPGRAM_API_KEY = os.getenv('DEEPGRAM_API_KEY')
FILE_PATH = 'path/to/your/audiofile.wav'

deepgram = Deepgram(DEEPGRAM_API_KEY)

async def transcribe_audio():
    try:
        with open(FILE_PATH, 'rb') as audio:
            source = {'buffer': audio, 'mimetype': 'audio/wav'}
            response = await deepgram.transcription.prerecorded(source)
            print('Transcription:', response)
    except Exception as e:
        print('Error:', e)

asyncio.run(transcribe_audio())

File: code/community/1304605478935859208/deepgram-speech-to-text-aws-lambda-websockets.js
const fs = require('fs');
const { Deepgram } = require('@deepgram/sdk');

const deepgramApiKey = process.env.DEEPGRAM_API_KEY;
const deepgram = new Deepgram(deepgramApiKey);

const file = fs.readFileSync('path/to/your/audiofile.wav');

async function transcribeAudio() {
  try {
    const response = await deepgram.transcription.preRecorded({
      buffer: file,
      mimetype: 'audio/wav',
    });
    console.log('Transcription:', response.results.channels);
  } catch (error) {
    console.error('Error:', error);
  }
}

transcribeAudio();
